src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
-        jsondata = json.dumps(credentials)
+        Retrieve a session token.
 
+        Args:
+            email: Email of user
+            password: Password of the user
+        """
         response = requests.post(
-            self._server + "/v2/authentication",
+            self.server + "/v2/authentication",
             headers={"Content-Type": "application/json; charset=UTF-8"},
-            data=jsondata,
+            data=json.dumps({"email": email, "password": password}),
             timeout=20,
         )
         check_for_api_error(response)
         result = response.json()
-        self._token = result["token"]
-
-    def get_token(self) -> str:
-        """
-        Returns the token
-        :return: token string
-        """
-
-        return self._token
-
-    @property
-    def token(self) -> str:
-        """Get the token of this connection"""
-        return self._token
+        self.token = result["token"]
 
     def start_logging(self) -> None:
         """Start writing every API call to a file"""
-        self._log = True
+        self.log = True
+        self.log_directory.mkdir(exist_ok=True)
 
-    def stop_logging(self):
+    def stop_logging(self) -> None:
         """Stop writing every API call to a file"""
-        self._log = False
+        self.log = False
 
     def logout(self) -> None:
-        """
-        Performs a logout
-        :return: None
-        """
-
-        if self._token is not None:
+        """Delete the session token"""
+        if self.token:
             response = requests.delete(
-                self._server + "/v2/authentication",
-                headers={"Authorization": "Bearer " + self._token},
+                self.server + "/v2/authentication",
+                headers={"Authorization": "Bearer " + self.token},
                 timeout=20,
             )
             check_for_api_error(response)
-            self._token = None
-
-    def __del__(self):
-        pass
-        # self.logout()
-
-    def post(self, path: str, jsondata: Optional[str] = None):
+            self.token = None
+
+    def write_log_file(
+        self,
+        method: str,
+        route: str,
+        headers: dict[str, str],
+        jsondata: Optional[str],
+        params: Optional[dict[str, Any]],
+        response: requests.Response,
+    ) -> None:
         """
-        Post Json data to a given server using a HTTP POST request
-        :param path: Path of RESTful route
-        :param jsondata: Valid JSON as string
-        :return: Response from server
+        If logging is enabled (self.log == True), write the request and response to a file.
+
+        Args:
+            method: HTTP method (POST, GET, PUT, DELETE)
+            route: route of DSP-API that was called
+            headers: headers of the HTTP request
+            jsondata: data sent to the server
+            response: response of the server
         """
+        if not self.log:
+            return
```

Comment:
I would check this condition outside of the function: like this, the function does two things (and should be renamed to `write_log_file_or_maybe_not()` 😉  ), but it should only do one thing.

Reply:
good point. done